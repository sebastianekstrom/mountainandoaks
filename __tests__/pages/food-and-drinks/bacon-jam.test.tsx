import React from "react";
import { render, screen } from "@testing-library/react";
import BaconJam from "pages/food-and-drinks/bacon-jam";

describe("BaconJam page", () => {
  it("renders title correctly", () => {
    render(<BaconJam />);
    const title = screen.queryAllByText(/Bacon jam/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<BaconJam />);
    const ingress = screen.queryAllByText(/TODO./i);
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<BaconJam />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });
});
