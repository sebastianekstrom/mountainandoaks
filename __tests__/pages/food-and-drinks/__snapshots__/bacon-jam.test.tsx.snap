// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaconJam page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Bacon
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          280
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Onion
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Shallots
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Maple syrup
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          50
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Brown sugar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          40
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Apple cider vinegar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          35
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Coffee
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          15
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Garlic
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          10
          g
        </p>
      </li>
    </ul>
  </div>
</section>
`;
